File: support_tools/swigwin-3.0.12/Examples/test-suite/go/exception_order_runme.go
package main

import "strings"
import . "./exception_order"

func main() {
	a := NewA()

	func() {
		defer func() {
			e := recover()
			if strings.Index(e.(string), "E1") == -1 {
				panic(e.(string))
			}
		}()
		a.Foo()
	}()

	func() {
		defer func() {
			e := recover()
			if strings.Index(e.(string), "E2") == -1 {
				panic(e.(string))
			}
		}()
		a.Bar()
	}()

	func() {
		defer func() {
			e := recover()
			if e.(string) != "postcatch unknown" {
				panic("bad exception order")
			}
		}()
		a.Foobar()
	}()

	func() {
		defer func() {
			e := recover()
			if strings.Index(e.(string), "E1") == -1 {
				panic(e.(string))
			}
		}()
		a.Barfoo(1)
	}()

	func() {
		defer func() {
			e := recover()
			if strings.Index(e.(string), "E2") == -1 {
				panic(e.(string))
			}
		}()
		a.Barfoo(2)
	}()
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/exception_order_runme.py
from exception_order import *

# This test is expected to fail with -builtin option.
# Throwing builtin classes as exceptions not supported
if is_python_builtin():
    exit(0)

a = A()

try:
    a.foo()
except E1, e:
    pass
except:
    raise RuntimeError, "bad exception order"

try:
    a.bar()
except E2, e:
    pass
except:
    raise RuntimeError, "bad exception order"

try:
    a.foobar()
except RuntimeError, e:
    if e.args[0] != "postcatch unknown":
        print "bad exception order",
        raise RuntimeError, e.args


try:
    a.barfoo(1)
except E1, e:
    pass
except:
    raise RuntimeError, "bad exception order"

try:
    a.barfoo(2)
except E2, e:
    pass
except:
    raise RuntimeError, "bad exception order"
